### PR TITLE
TST: Address matplotlib 3.10 deprecation of vert=

### PR DESCRIPTION
**Pull request by mroeschke**
e.g. https://app.circleci.com/jobs/github/pandas-dev/pandas/243601

### CODE REVIEW

**Comment by mroeschke** [comment]
Merging to get a green CI

File: pandas/tests/plotting/frame/test_frame.py [hunk 1]
```diff
@@ -1070,41 +1070,66 @@ def test_boxplot_series_positions(self, hist_df):
...
         labels = [pprint_thing(c) for c in numeric_cols]
 
         # if horizontal, yticklabels are rotated
-        ax = df.plot.box(rot=50, fontsize=8, vert=False)
+        kwargs = (
+            {"vert": False}
+            if Version(mpl.__version__) < Version("3.10")
+            else {"orientation": "horizontal"}
+        )
+        ax = df.plot.box(rot=50, fontsize=8, **kwargs)
```
Comment (jorisvandenbossche) [inline pandas/tests/plotting/frame/test_frame.py:1089]: Just to understand the change: even though using the new keyword for mpl >= 3.10, it is still failing somehow? (given you added the xfail)

**Comment by jorisvandenbossche** [comment]
Thanks for addressing this!

File: pandas/tests/plotting/frame/test_frame.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/plotting/frame/test_frame.py:1089]: Yeah that's correct. It appears that changing this to use the equivalent, new matplotlib 3.10 keyword causes some tests to fail. I was unable to determine if the failure was on our side or the matplotlib side

**Pull request merged**